juju/charm.py

```diff
@@ -1,62 +1,70 @@
 from juju.framework import Object, Event, EventBase, EventsBase
 
 
-class InstallEvent(EventBase):
+class JujuEvent(EventBase):
     pass
 
-class StartEvent(EventBase):
+
+class JujuHookEvent(JujuEvent):
```

Comment:
Why do we need JujuHookEvent as opposed to just using JujuEvent? JujuHookEvent is not used in `main.py` with this change so I think this isn't needed.

I'd avoid having "Hook" in the name as well because it refers to an implementation detail of how a unit agent executes files called "hooks".

Reply:
I mentioned in the PR comment above why I included both: in case we need to distinguish between, say, action events and hook events in the future, which I strongly suspect we'll need to do since they'll need to be wired up differently by the framework.